src/core/protocol-authorization.ts

```diff
@@ -315,14 +364,22 @@ export class ProtocolAuthorization {
     const roleRecordFilter: Filter = {
       interface         : DwnInterfaceName.Records,
       method            : DwnMethodName.Write,
-      protocol          : recordsWrite.message.descriptor.protocol!,
+      protocol          : protocolUri,
       protocolPath      : protocolRole,
       recipient         : incomingMessage.author!,
       isLatestBaseState : true,
     };
+
     if (roleRuleSet?.$contextRole) {
```

Comment:
Gentle reminder there are still a few places with annoying uncovered branch coverage.

Reply:
```
protocol-authorization.ts      |   99.36 |    95.14 |     100 |   99.36 | 272-273,609-610
```
I don't see this line in the uncovered lines when I run `npm run test:node`. Lines `272-273,609-610` (in commit [49d0cdd](https://github.com/TBD54566975/dwn-sdk-js/pull/560/commits/49d0cdd78de9e281054effed3ee9eda488216645)) have comments explaining why they are uncovered. Are you getting different test coverage results than me?

Reply:
https://app.codecov.io/github/TBD54566975/dwn-sdk-js/blob/main/src%2Fcore%2Fprotocol-authorization.ts shows what I am seeing.

Reply:
Pushed an additional test that should catch this.

Reply:
🙏